### ISSUE TITLE
update from 4.0.5/4.0.6 to 4.1.1/4.1.2/4.2.0 does not work

### ISSUE THREAD

**Issue by zwnk** [RocketChat/Rocket.Chat#23660]
##########################################################################################

> Hello, thanks to @lightweight [#23836 (comment)](https://github.com/RocketChat/Rocket.Chat/issues/23836#issuecomment-1049258909) i have been able to upgrade my instance to 4.6.0. Just disable "Direct Reply Enable" in "Email" section in Rocket.Chat administration settings.

This was the soltuion.

##########################################################################################

### Steps to reproduce:
Pull 4.1.1 rocket.chat image
stop rc container
rm rc container 
start with new image

### Expected behavior:
it starts without an error

### Actual behavior:
```
LocalStore: store created at
LocalStore: store created at
LocalStore: store created at
Error occurred ...
meteor://💻app/app/app.js:193807
      throw err;
      ^

Error: 139918009431936:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:

    at Socket.ondata (internal/js_stream_socket.js:77:22)
    at Socket.emit (events.js:314:20)
    at Socket.EventEmitter.emit (domain.js:483:12)
    at addChunk (_stream_readable.js:297:12)
    at readableAddChunk (_stream_readable.js:272:9)
    at Socket.Readable.push (_stream_readable.js:213:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:188:23)
    at TCP.callbackTrampoline (internal/async_hooks.js:126:14) {
  library: 'SSL routines',
  function: 'ssl3_get_record',
  reason: 'wrong version number',
  code: 'ERR_SSL_WRONG_VERSION_NUMBER',
  source: 'socket'
}
Could not start Rocket.Chat. Waiting 5 secs...
```

### Server Setup Information:

- Version of Rocket.Chat Server: 4.0.5 to 4.1.1
- Operating System: ubuntu 20.04
- Deployment Method: docker-compose
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version:  the one that comes with the image
- MongoDB Version: 5.0

Nginx as reverse proxy with standard rchat config. rocket chat app is bound to 127.0.0.1:3000
[like shown here](https://docs.rocket.chat/quick-start/installing-and-updating/docker-containers)

here the used nginx conf:
```
server {
    listen 80;
	listen [::]:80;
    server_name domain.com;
    return 301 https://$host$request_uri;
}

server {
    listen 443 ssl;
    listen [::]:443 ssl;
    ssl_certificate /etc/letsencrypt/live/domain.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/domain.com/privkey.pem; # managed by Certbot
    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_prefer_server_ciphers on;
    ssl_session_cache shared:SSL:50m;
    ssl_session_timeout 1d;
    ssl_session_tickets off;
    
    # For the federation port
    #listen 8448 ssl http2 default_server;
    #listen [::]:8448 ssl http2 default_server;

    server_name domain.com;

    location / {
            proxy_pass http://127.0.0.1:3000/;
	    client_max_body_size 50M;
            proxy_http_version 1.1;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "upgrade";
            proxy_set_header Host $http_host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto https;
            proxy_set_header X-Nginx-Proxy true;
            proxy_redirect off;
    }
}
```

here the docker-compose.yml:
```
version: "3.7"

services:
  rocketchat:
    image: registry.rocket.chat/rocketchat/rocket.chat:4.0.6
#$    image: rocketchat
    command: >
      bash -c
        "for (( ; ; )); do
          node main.js &&
          s=$$? && break || s=$$?;
          echo \"Could not start Rocket.Chat. Waiting 5 secs...\";
          sleep 5;
        done; (exit $$s)"
    restart: on-failure
    volumes:
      - ./uploads:/app/uploads
    environment:
      - PORT=3000
      - ROOT_URL=https://my.domain.com
      - MONGO_URL=mongodb://mongo:27017/rocketchat
      - MONGO_OPLOG_URL=mongodb://mongo:27017/local
      - Accounts_UseDNSDomainCheck=false
    depends_on:
      - mongo
    ports:
      - 3000:3000
    networks:
      - net
      - rocket

  mongo:
    image: mongo:5.0
    restart: on-failure
    volumes:
      - /opt/rocketchat_db/data/db:/data/db
      - ./data/dump:/dump
    command: >
      bash -c
        "while [ ! -f /data/db/WiredTiger ]; do
          echo \"wiredTiger migration hasn't started yet. Waiting 30 secs...\";
          sleep 30;
        done;
        docker-entrypoint.sh mongod --oplogSize 128 --replSet rs0 --storageEngine=wiredTiger;"
    depends_on:
      - migrator
    networks:
      - rocket

  migrator:
    restart: on-failure
    build: ./nmap2wt/docker/
    volumes:
       - /opt/rocketchat_db/data/db:/data/db
#      - ./data/db:/data/db
    networks:
      - rocket
 mongo-init-replica:
    image: mongo:5.0
    command: >
      bash -c
        "for (( ; ; )); do
          mongo mongo/rocketchat --eval \"
            rs.initiate({
              _id: 'rs0',
              members: [ { _id: 0, host: 'localhost:27017' } ]})\" &&
          s=$$? && break || s=$$?;
          echo \"Could not reach MongoDB. Waiting 5 secs ...\";
          sleep 5;
        done; (exit $$s)"
    depends_on:
      - mongo
    networks:
      - rocket
  # hubot, the popular chatbot (add the bot user first and change the password before starting this image)
  hubot:
    image: rocketchat/hubot-rocketchat:latest
    restart: unless-stopped
    environment:
      - ROCKETCHAT_URL=rockechat:3000
      - ROCKETCHAT_ROOM=GENERAL
      - ROCKETCHAT_USER=bot
      - ROCKETCHAT_PASSWORD=thepassword
      - BOT_NAME=dmclbot
  # you can add more scripts as you'd like here, they need to be installable by npm
      - EXTERNAL_SCRIPTS=hubot-help,hubot-seen,hubot-links,hubot-diagnostics
    depends_on:
      - rocketchat
    labels:
      - "traefik.enable=false"
    volumes:
      - ./scripts:/home/hubot/scripts
  # this is used to expose the hubot port for notifications on the host on port 3001, e.g. for hubot-jenkins-notifier
    ports:
      - 3001:8080

networks:
  net:
    external: true
  rocket:
    internal: true
```

**Comment by TheDigitalEagle** [comment]
Looks to me like you're trying to use an old SSL method or there's an issue with your certificate.

I strongly recommend you use a reverse proxy with nginx, then you won't have to worry about what version of ssl implemention you're trying to support.

**Comment by husim0** [comment]
Same problem here. I'm using a reverse proxy with traefik. I'm not using anything related to SSL in my rocketchat app container.

**Comment by zwnk** [comment]
yes, i'm using nginx as a reverse proxy.

**Comment by zwnk** [comment]
my nginx config just uses tls 1.2 atm
```
ssl_protocols TLSv1.2;
```

i will change it to:
```
ssl_protocols TLSv1.2 TLSv1.3;
```
and report back.

**Comment by zwnk** [comment]
same error. nothing changed.

**Comment by zwnk** [comment]
same error on 4.1.2 .........

**Comment by zwnk** [comment]
no further comment on this one?

**Comment by zwnk** [comment]
i tried to update to 4.0.6 and it worked, the docker image wasn't avaiable for this version for some time.
next i tried to update to 4.1.2 and 4.2.0. 
Same error as usual?!

**Comment by zwnk** [comment]
> Same problem here. I'm using a reverse proxy with traefik. I'm not using anything related to SSL in my rocketchat app container.

did you get your update working?

**Comment by husim0** [comment]
> > Same problem here. I'm using a reverse proxy with traefik. I'm not using anything related to SSL in my rocketchat app container.
> 
> did you get your update working?

I successfully upgraded from 4.0.5 to 4.0.6 but I didn't tried to upgrade to 4.2.0 yet. I'm waiting for 4.2.1 for re-trying. But I'm a bit worried of any db schema upgrade so I think I would try it on a backup because I don't know how db downgrade is managed if necessary.

**Comment by zwnk** [comment]
> 

i tried to upgrade to 4.2.0 yesterday but the same error occurs.
just switched back to 4.0.6 without any issue.

**Comment by husim0** [comment]
No Christmas miracle for that issue ? Still cannot update from 4.0.6 to anything higher

**Comment by zwnk** [comment]
@husim0 did you update your mongo db to a version >4.0?

**Comment by husim0** [comment]
Yes, I'm currently using mongodb 4.2.17 / wiredTiger (oplog Activé)

**Comment by zwnk** [comment]
i updated to 5.0.3 / wiredTiger. 
Last week i did some tests. If you leave the db on 4.0 (docker-compose.yml "default") you can update the rc instance.

**Comment by zwnk** [comment]
i tried to update to 4.4.2
same error ...
i tried http only nginx, same error. i don't understand what is causing the issue.

**Comment by lightweight** [comment]
This needs the attention of someone with deep MongoDB and Rocket.Chat knowledge, e.g. on the core team. I still haven't upgraded from 4.0.6... I'm getting pretty frustrated. This seems to be affecting a lot of people. I went to quite a bit of trouble to upgrade my MongoDBs from 4.0 to 4.4 as instructed, but now can't upgrade R.C...

**Comment by zwnk** [comment]
@lightweight you are having the same issue?

**Comment by lightweight** [comment]
Yup. I reported it #23836

**Comment by zwnk** [comment]
I will take a look if I can downgrade from 5 to 4.x.

**Comment by husim0** [comment]
I'm using mongodb 4.2 and can't upgrade. Indeed the failure occured soon after my mongodb upgrade from 4.0 to 4.2

**Comment by husim0** [comment]
I downgraded today from 4.2 to 4.0 and updated Rocketchat to latest 4.6.0 and I still have the same error message `SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332`

**Comment by husim0** [comment]
My Rocket.Chat instance is in production and I'm beginning to be afraid about security. Would be nice to have a feedback from Rocket.Chat team as we are unable to upgrade since...november 2021.

**Comment by zwnk** [comment]
i don't think that someone will help. 
do you know a way to export the mongodb and import it to a fresh setup of rocket chat? i'm not very familiar with mongodb.
maybe there isn't a way for me since i'm already on 5.0 and the default version is 4.0.

**Comment by husim0** [comment]
I'm not so sure our issue is related to MongoDB after all. Actually last version of Rocket.Chat is supposed to support MongoDB 3.6, 4.0, 4.2, 4.4 and 5.0, so is RC version 4.0.6. I tried to downgrade and the result is the same. Not sure you can export a MongoDB dump from 5.0 and import it to 4.0.

I think there is something else involved here.

**Comment by zwnk** [comment]
i setup a new instance on the same host under the same domain (during off hours) and had no issues. so it is nothing related to the nginx reverse proxy running on the machine. 
the only thing i did on version 4.0.5 was upgrading mongodb with the tut provided by rocket chat.
i also don't use any fancy setup parameters, its just a small chat server for 10 people.

**Comment by husim0** [comment]
Hmm, okay, thank you for your feedback. I think it must be related to something particular to our two instances as we seem to be the only one affected :-/

**Comment by husim0** [comment]
Hello, thanks to @lightweight  https://github.com/RocketChat/Rocket.Chat/issues/23836#issuecomment-1049258909 i have been able to upgrade my instance to 4.6.0. Just disable "Direct Reply Enable" in "Email" section in Rocket.Chat administration settings.
